File: app/src/main/java/ru/alexsuvorov/paistewiki/model/News.java
package ru.alexsuvorov.paistewiki.model;

import android.arch.persistence.room.ColumnInfo;
import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Ignore;
import android.arch.persistence.room.Index;
import android.arch.persistence.room.PrimaryKey;
import android.support.annotation.NonNull;

import lombok.Getter;
import lombok.Setter;

@Entity(tableName = "news_table", indices = @Index(value = {"news_url"}, unique = true))
public class News {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "news_id")
    @Getter @Setter @NonNull
    public long news_id = 1;
    @ColumnInfo(name = "news_title")
    @Getter @Setter
    public String title;
    @ColumnInfo(name = "news_category")
    @Getter @Setter
    public String category;
    @ColumnInfo(name = "news_url")
    @Getter @Setter
    public String url;
    @ColumnInfo(name = "news_index")
    @Getter @Setter
    public long news_index;

    public News() {
    }

    @Ignore
    public News(@NonNull long news_id, String title, String category, String url, long news_index) {
        this.news_id = news_id;
        this.title = title;
        this.category = category;
        this.url = url;
        this.news_index = news_index;
    }
}

File: app/src/main/java/ru/alexsuvorov/paistewiki/model/News.kt
package ru.alexsuvorov.paistewiki.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(tableName = "news_table", indices = [Index(value = ["news_url"], unique = true)])
class News {
    @JvmField
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "news_id")
    var news_id: Long = 1

    @JvmField
    @ColumnInfo(name = "news_title")
    var title: String? = null

    @JvmField
    @ColumnInfo(name = "news_category")
    var category: String? = null

    @JvmField
    @ColumnInfo(name = "news_url")
    var url: String? = null

    @JvmField
    @ColumnInfo(name = "news_index")
    var news_index: Long = 0

    constructor()

    @Ignore
    constructor(news_id: Long, title: String?, category: String?, url: String?, news_index: Long) {
        this.news_id = news_id
        this.title = title
        this.category = category
        this.url = url
        this.news_index = news_index
    }
}